controller/controller.go

```diff
@@ -101,9 +106,12 @@ func (ctrl *baseController) init(ctx context.Context) error {
 		}
 
 		units = append(units, unit{
-			taskName: task.Name,
-			template: template,
-			driver:   d,
+			taskName:  task.Name,
+			template:  template,
+			driver:    d,
+			providers: task.ProviderNames(),
+			services:  task.ServiceNames(),
+			source:    task.Source,
```

Comment:
This is starting to seem like a copy of a task, we're pulling so much from it. At the moment I think it is fine but we might want to review this in the future to see if we want to keep copying all these values or just use the task itself.

Reply:
:nod: I debated just using task as a field but didn't feel quite ready. Agree, we'll keep an eye on this and switch over to task if another field needs to be added.